### PR TITLE
Anthropic Examples/Readme Guide Created and Proper Reference Noted.

### LINKED ISSUE #446: [Docs]: Anthropic support docs
### 📘 Current State of Documentation

We currently don't show how to use the Anthropic SDK in the docs. Update 2 things:
1. Docs page
2. Readme 

### 📖 Suggested Improvement

- Sample code snippet in docs
- Sample code snippet in readme
- Notebook example

### 🔗 Affected Documentation Pages

_No response_

### 🔍 Additional Context

_No response_

### 🤝 Contribution

- [ ] Yes, I'd be happy to submit a pull request with these changes.
- [ ] I need some guidance on how to contribute.
- [ ] I'd prefer the Agentops team to handle this update.

### PR DESCRIPTION
**Pull request by TKTSWalker**
## 📥 Pull Request

**📘 Description**
_Changed link on Anthropic example to an anthropic readme I created; aside from the pre existing barebones example, I also included three examples showing how the async, sync and tool function could be used. 

The sync example shows a case where a sentence generated by simple mixing and connecting is given as context to create a story. The async example has the LLM generate a script line based off the game "TitanFall", depending on the state of the Titan and the personality. At the same time, 4 UUIDs are generated on async as a simple example. Finally, the tool function gives a piece of "Cyberware" based off the user's requested company. Then with the job role the user wants, the LLM says if it is a good or bad fit.

I purposefully made the examples connect to different media as when starting out, lesser experienced developers often have issues following along without a greater level of explanation and focus is an issue as well. By connecting it to better known medias, it remains more engaging while allowing the reader to possibly create parallels between the instructions being shown and the base concepts they're already aware of.

I am also requesting that I am allowed to use AgentOps to show how to do a few things such as setting up a history to keeping it small but with context through summarization when needed. Then, a few use case examples could be shown to help newer developers both understand how to develop with LLMs and how to do so with AgentOps!_

**🧪 Testing**
_Ran journals on Kaggle piece by piece (Which is why the async example is changed so much aside from having a little more meaningful example) before sending it over to a friend to test as well._

### CODE REVIEW

**Comment by areibman** [comment]
Let's take a look @the-praxs

File: examples/anthropic_examples/anthropic-example-async.ipynb
Comment (areibman) [inline examples/anthropic_examples/anthropic-example-async.ipynb:1]: Are you trying to demonstrate Anthropic computer use? These notebooks unfortunately do not work

```
---------------------------------------------------------------------------
BadRequestError                           Traceback (most recent call last)
Cell In[10], line 1
----> 1 stream = client.messages.create(
      2     max_tokens=2400, 
      3     model="claude-3-5-sonnet-20240620",  # Comma added here
      4     messages=[
      5         {
      6             "role": "user",
      7             "content": "Create a story based on the three words given to you by the computer."
      8         },
      9         {
     10             "role": "computer",
     11             "content": "{A foolish doll} {died in a world} {of ended dreams.}"
     12         },
     13         {
     14             "role": "assistant",
     15             "content": defaultstory
     16         },
     17         {
     18             "role": "user",
     19             "content": "Create a story based on the three words given to you by the computer."
     20         },
     21         {
     22             "role": "computer",
     23             "content": generatedsentence
     24         }
     25     ],
     26     stream=True,
     27 )
     29 response = ""
     30 for event in stream:

File ~/anaconda3/lib/python3.11/site-packages/anthropic/_utils/_utils.py:274, in required_args.<locals>.inner.<locals>.wrapper(*args, **kwargs)
    272             msg = f"Missing required argument: {quote(missing[0])}"
    273     raise TypeError(msg)
--> 274 return func(*args, **kwargs)

File ~/anaconda3/lib/python3.11/site-packages/anthropic/resources/messages.py:878, in Messages.create(self, max_tokens, messages, model, metadata, stop_sequences, stream, system, temperature, tool_choice, tools, top_k, top_p, extra_headers, extra_query, extra_body, timeout)
    871 if model in DEPRECATED_MODELS:
    872     warnings.warn(
    873         f"The model '{model}' is deprecated and will reach end-of-life on {DEPRECATED_MODELS[model]}.\nPlease migrate to a newer model. Visit https://docs.anthropic.com/en/docs/resources/model-deprecations for more information.",
    874         DeprecationWarning,
    875         stacklevel=3,
    876     )
--> 878 return self._post(
    879     "/v1/messages",
    880     body=maybe_transform(
    881         {
    882             "max_tokens": max_tokens,
    883             "messages": messages,
    884             "model": model,
    885             "metadata": metadata,
    886             "stop_sequences": stop_sequences,
    887             "stream": stream,
    888             "system": system,
    889             "temperature": temperature,
    890             "tool_choice": tool_choice,
    891             "tools": tools,
    892             "top_k": top_k,
    893             "top_p": top_p,
    894         },
    895         message_create_params.MessageCreateParams,
    896     ),
    897     options=make_request_options(
    898         extra_headers=extra_headers, extra_query=extra_query, extra_body=extra_body, timeout=timeout
    899     ),
    900     cast_to=Message,
    901     stream=stream or False,
    902     stream_cls=Stream[RawMessageStreamEvent],
    903 )

File ~/anaconda3/lib/python3.11/site-packages/anthropic/_base_client.py:1260, in SyncAPIClient.post(self, path, cast_to, body, options, files, stream, stream_cls)
   1246 def post(
   1247     self,
   1248     path: str,
   (...)
   1255     stream_cls: type[_StreamT] | None = None,
   1256 ) -> ResponseT | _StreamT:
   1257     opts = FinalRequestOptions.construct(
   1258         method="post", url=path, json_data=body, files=to_httpx_files(files), **options
   1259     )
-> 1260     return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))

File ~/anaconda3/lib/python3.11/site-packages/anthropic/_base_client.py:937, in SyncAPIClient.request(self, cast_to, options, remaining_retries, stream, stream_cls)
    928 def request(
    929     self,
    930     cast_to: Type[ResponseT],
   (...)
    935     stream_cls: type[_StreamT] | None = None,
    936 ) -> ResponseT | _StreamT:
--> 937     return self._request(
    938         cast_to=cast_to,
    939         options=options,
    940         stream=stream,
    941         stream_cls=stream_cls,
    942         remaining_retries=remaining_retries,
    943     )

File ~/anaconda3/lib/python3.11/site-packages/anthropic/_base_client.py:1041, in SyncAPIClient._request(self, cast_to, options, remaining_retries, stream, stream_cls)
   1038         err.response.read()
   1040     log.debug("Re-raising status error")
-> 1041     raise self._make_status_error_from_response(err.response) from None
   1043 return self._process_response(
   1044     cast_to=cast_to,
   1045     options=options,
   (...)
   1049     retries_taken=options.get_max_retries(self.max_retries) - retries,
   1050 )

BadRequestError: Error code: 400 - {'type': 'error', 'error': {'type': 'invalid_request_error', 'message': 'messages: Unexpected role "computer". Allowed roles are "user" or "assistant"'}}
```

File: examples/anthropic_examples/antrophic-example-tool.ipynb
Comment (areibman) [inline examples/anthropic_examples/antrophic-example-tool.ipynb:1]: See comment about computer use

**Comment by areibman** [comment]
Thanks @TKTSWalker! The sync notebook looks great but it looks like you're using Anthropic Computer Use in the others, and those don't work. Is that what you're after? Or did you just coincidentally name the roles "computer"

**Comment by TKTSWalker** [comment]
Good afternoon! I named the role computer by coincidence, the name should be able to be replaced with a term like "Machine" if need be! i'm just used to naming the role computer, as with the way the system works, the output shouldn't be affected! 

If there are any problems with this however, please feel free to let me know and I will act accordingly!

**Comment by areibman** [comment]
> Good afternoon! I named the role computer by coincidence, the name should be able to be replaced with a term like "Machine" if need be! i'm just used to naming the role computer, as with the way the system works, the output shouldn't be affected!
> 
> If there are any problems with this however, please feel free to let me know and I will act accordingly!

Hahaha that's what I figured. Can you make the changes and re-run the notebooks end-to-end to make sure they run properly? Will merge ASAP when you push it

**Comment by TKTSWalker** [comment]
Hello, it's no problem! Going to start doing so in a moment, I'll share a
copy with all the output logs once checked!

**Comment by TKTSWalker** [comment]
Whoops, found my errors! I was thinking about raw text input/output like the Llama.cpp instead of accounting for the fact that anthropic deadlocks assistant and user!

Also there are one or two errors but the reason it may not have worked on your end aside from roles is the fact that I was supposed to put client = Anthropic(api_key=ANTHROPIC_API_KEY) but accidently removed everything in the ()!


Fixing now and I already got sync working again, it should take a few minutes at least and an hour or two at most (considering I don't get sidetracked)!
image_2024-11-09_105434685

**Comment by TKTSWalker** [comment]
Hello @areibman it's all set! I had to handle a few things related to revamping as I realized I made some glaring mistakes that barely kept things working! You can check

https://github.com/TKTSWalker/agentops/blob/main/examples/anthropic_examples/anthropic-example-sync.ipynb
https://github.com/TKTSWalker/agentops/blob/main/examples/anthropic_examples/anthropic-example-async.ipynb
https://github.com/TKTSWalker/agentops/blob/main/examples/anthropic_examples/antrophic-example-tool.ipynb

to see that these were ran and the outputs are still there to view to confirm as working!

Besides that, I do believe that the way listed for the tool functions do not work in the default example and want to handle this but it will take a while to properly implement it! For example, I had problems while keeping stream on! Also, the output itself is also not a string but rather a sort of weirdish output (kind of similar to Json)! 

Even if there is a way to get things from text, the JSON format is extremely valuable as it allows for easily getting the AI's text and related tools!

I am completely sure that I can make a system around automatically handling this but it would definitely take a little time and direct focus so i'm hoping to do so in a different push request entirely!

**Review by areibman — APPROVED** [review]

**Comment by areibman** [comment]
Merged-- thanks @TKTSWalker !!

**Comment by TKTSWalker** [comment]
Hey, no problem! Moving onto the next!

**Pull request merged**